src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildInfoAction.java

```diff
@@ -0,0 +1,41 @@
+package org.jenkinsci.plugins.workflow.support.steps.build;
+
+import hudson.model.InvisibleAction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class BuildInfoAction extends InvisibleAction {
+    private final List<BuildInfo> buildInfoList = new ArrayList<>();
+
+    public BuildInfoAction(String projectName, int buildNumber) {
+        buildInfoList.add(new BuildInfo(projectName, buildNumber));
+    }
+
+    public List<BuildInfo> getBuildInfoList() {
+        return buildInfoList;
+    }
+
+    public void addBuildInfo(String projectName, int buildNumber) {
+        buildInfoList.add(new BuildInfo(projectName, buildNumber));
+    }
+
+    static class BuildInfo {
+        private final String projectName;
+        private final int buildNumber;
+
+        public BuildInfo(String projectName, int buildNumber) {
```

Comment:
How can I prevent this?
Btw, is it possible that user has access to list of actions of some build, but doesn't have access to the log of this build? (right now plugin publishes names/build numbers of triggered jobs to the log).

For now I made this class private and only list of runs can be retrieved from action. I hope that it would be safer.